util/process_perf_data.py

```diff
@@ -282,14 +288,17 @@ def post_treeherder_jobs(client, revisions, s3=None):
             print "Failed to generate data for %s: %s, probably still running" % (revision, e)
             continue
 
-        # NB: In theory we could batch these, but each collection has to be from
-        #     the same repo and it's possible we have different repos in our
-        #     dataset.
-        client.post_collection(repo, tjc)
-        #print tjc.to_json()
-
-        successful.append(revision)
-        print "Submitted perf data for %s to %s" % (revision, client.host)
+        try:
+            # NB: In theory we could batch these, but each collection has to be from
+            #     the same repo and it's possible we have different repos in our
+            #     dataset.
+            client.post_collection(repo, tjc)
+            #print tjc.to_json()
+
+            successful.append(revision)
+            print "Submitted perf data for %s to %s" % (revision, client.host)
+        except Exception as e:
+            print "Failed to submit data for %s: %s" % (revision, e)
```

Comment:
It won't get added to the successful list, so the next call to `process_perf_data` will attempt to resubmit. That's currently configured to run every 5 minutes.